core-primitives/top-pool-author/src/author.rs

```diff
@@ -83,15 +88,17 @@ where
 	ocall_api: Arc<OCallApi>,
 }
 
-impl<TopPool, TopFilter, StateFacade, ShieldingKeyRepository, OCallApi>
-	Author<TopPool, TopFilter, StateFacade, ShieldingKeyRepository, OCallApi>
+impl<TopPool, TopFilter, StateFacade, ShieldingKeyRepository, OCallApi, TCS, G>
+	Author<TopPool, TopFilter, StateFacade, ShieldingKeyRepository, OCallApi, TCS, G>
 where
-	TopPool: TrustedOperationPool + Sync + Send + 'static,
-	TopFilter: Filter<Value = TrustedOperation>,
+	TopPool: TrustedOperationPool<StfTrustedOperation<TCS, G>> + Sync + Send + 'static,
+	TopFilter: Filter<Value = StfTrustedOperation<TCS, G>>,
 	StateFacade: QueryShardState,
 	ShieldingKeyRepository: AccessKey,
 	<ShieldingKeyRepository as AccessKey>::KeyType: ShieldingCryptoDecrypt,
 	OCallApi: EnclaveMetricsOCallApi + Send + Sync + 'static,
+	TCS: PartialEq + Encode + Clone + Debug + Send + Sync,
+	G: PartialEq + Encode + Clone + PoolTransactionValidation + Debug + Send + Sync,
```

Comment:
we have that in a lot of places now. It may be worth simplifying this at some point, but I first wanted to see it work. Its quite some redundancy....so we might rather define a type argument for the overall top and aggregate trait bounds